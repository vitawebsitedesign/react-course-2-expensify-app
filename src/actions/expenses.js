import {uuid} from 'uuidv4';

export const addExpense = ({
    description = '',
    note = '',
    amt = 0,
    createdAt = new Date()
}) => ({
    type: 'ADD_EXPENSE',
    expense: {
        id: uuid(),
        description,
        note,
        amt,
        createdAt
    }
});

export const removeExpense = (id) => ({
    type: 'REMOVE_EXPENSE',
    id
});

export const editExpense = (id, expense) => ({
    type: 'EDIT_EXPENSE',
    id,
    expense
});
