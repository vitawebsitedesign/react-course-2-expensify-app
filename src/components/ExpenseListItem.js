import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({id, description, amt, createdAt}) => (
    <div>
        <h4>
            Description: {description}
            &nbsp;
            (<Link to={`/edit/${id}`}>Edit</Link>)
        </h4>
        <p>Amount: {numeral(amt).format('0,0.00')}</p>
        <p>Created at: {moment(createdAt).format('Do MMMM, YYYY')}</p>
    </div>
);

export default ExpenseListItem;

