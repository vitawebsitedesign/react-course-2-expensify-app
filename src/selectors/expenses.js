import moment from 'moment';

const getVisibleExpenses = (expenses, filters) => {
    const {text, sortBy, startDate, endDate} = filters;
    return expenses.filter(e => {
        const createdAtMoment = moment(e.createdAt);
        const startDateMatch = startDate ? createdAtMoment.isSameOrAfter(startDate, 'day') : true;
        const endDateMatch = endDate ? createdAtMoment.isSameOrBefore(endDate, 'day') : true;
        const textMatch = e.description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === 'date') {
            return a.createdAt - b.createdAt;
        } else if (sortBy === 'amt') {
            return b.amt - a.amt;
        }
        return 0;
    });
};

export default getVisibleExpenses;
