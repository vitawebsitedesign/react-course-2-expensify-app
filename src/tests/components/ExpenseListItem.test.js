import React from 'react';
import {shallow} from 'enzyme';
import {gum} from '../fixtures/expenses';
import ExpenseListItem from '../../components/ExpenseListItem';

test('should render expense list item', () => {
    const wrapper = shallow(<ExpenseListItem {...gum} />);
    expect(wrapper).toMatchSnapshot();
});
