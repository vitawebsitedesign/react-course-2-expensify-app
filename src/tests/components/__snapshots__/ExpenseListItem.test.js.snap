// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item 1`] = `
<div>
  <h4>
    Description: 
    gum
    Â  (
    <Link
      to="/edit/2"
    >
      Edit
    </Link>
    )
  </h4>
  <p>
    Amount: 
    99.00
  </p>
  <p>
    Created at: 
    16th April, 2020
  </p>
</div>
`;
